### ISSUE TITLE
GraphQL query fails for Interface.mac_address field

### ISSUE THREAD

**Issue by mzbroch** [nautobot/nautobot#3902]
### Environment
* Nautobot version (Docker tag too if applicable): ca5694aaa671 (v2.0.0b1) 
* Python version:
* Database platform, version:
* Middleware(s):

### Steps to Reproduce
1. Create GraphQL query

```
query ($device_id: ID!) {
        device(id: $device_id) {
          config_context
          hostname: name
          position
          serial
          primary_ip4 {
            id
            primary_ip4_for {
              id
              name
            }
          }
          tenant {
            name
          }
          tags {
            name
            slug
          }
          role {
            name
          }
          platform {
            name
            manufacturer {
              name
            }
            napalm_driver
          }
          location {
            name
            vlans {
              id
              name
              vid
            }
            vlan_groups {
              id
            }
          }
          interfaces {
            description
            mac_address
            enabled
            name
            ip_addresses {
              address
              tags {
                id
              }
            }
            connected_circuit_termination {
              circuit {
                cid
                commit_rate
                provider {
                  name
                }
              }
            }
            tagged_vlans {
              id
            }
            untagged_vlan {
              id
            }
            cable {
              termination_a_type
              status {
                name
              }
              color
            }
            tagged_vlans {
              location {
                name
              }
              id
            }
            tags {
              id
            }
          }
        }
      }
```

3. Execute GraphQL query
4.

### Expected Behavior

GraphQL for Interface.mac_address works

### Observed Behavior

GQL fails:

```
{
  "errors": [
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        0,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        1,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        2,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        3,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        4,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        5,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        6,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        7,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        8,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        9,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        10,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        11,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        12,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        13,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        14,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        15,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        16,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        17,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        18,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        19,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        20,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        21,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        22,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        23,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        24,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        25,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        26,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        27,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        28,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        29,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        30,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        31,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        32,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        33,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        34,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        35,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        36,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        37,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        38,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        39,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        40,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        41,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        42,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        43,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        44,
        "mac_address"
      ]
    },
    {
      "message": "Cannot return null for non-nullable field InterfaceType.mac_address.",
      "locations": [
        {
          "line": 44,
          "column": 13
        }
      ],
      "path": [
        "device",
        "interfaces",
        45,
        "mac_address"
      ]
    }
  ],
  "data": {
    "device": {
      "config_context": {},
      "hostname": "ams01-asw-01",
      "position": 34,
      "serial": "",
      "primary_ip4": null,
      "tenant": {
        "name": "IT Facility EMEA"
      },
      "tags": [],
      "role": {
        "name": "branch_access_switch"
      },
      "platform": {
        "name": "Cisco IOS",
        "manufacturer": {
          "name": "Cisco"
        },
        "napalm_driver": "ios"
      },
      "location": {
        "name": "ams01-server-room",
        "vlans": [],
        "vlan_groups": []
      },
      "interfaces": [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null
      ]
    }
  }
}
```

**Comment by bryanculver** [comment]
Looks like a miss from: https://github.com/nautobot/nautobot/pull/3046

Reproducable in a query as simple as:

```graphql
query {
  devices(limit: 10) {
    hostname: name
    interfaces {
      name
      mac_address
      ip_addresses {
        address
      }
    }
  }
}
```

**Comment by glennmatthews** [comment]
Fixed by #3950.
